docs/components/DataList/Web.stories.tsx

```diff
@@ -404,6 +408,309 @@ export const Basic: StoryObj<typeof DataList> = {
   ),
 };
 
+export const ClearAllFilters: StoryFn<typeof DataList> = args => {
+  interface SelectedFilters {
+    home: ComboboxOption[];
+    eyeColor: ComboboxOption[];
+  }
+
+  interface Filter {
+    key: keyof SelectedFilters;
+    label: string;
+    options: string[];
+    isSelected: boolean;
+    selectedFilters: ComboboxOption[];
+  }
+
+  type Entries<T> = {
+    [K in keyof T]: [K, T[K]];
+  }[keyof T][];
+
+  const selectedFiltersInitialState: SelectedFilters = {
+    home: [],
+    eyeColor: [],
+  };
+
+  const [selectedFilters, setSelectedFilters] = useState<SelectedFilters>(
+    selectedFiltersInitialState,
+  );
+
+  function removeAllFilters() {
+    setSelectedFilters(selectedFiltersInitialState);
+  }
+
+  function handleRemoveIndividualFilterGroup(type: keyof SelectedFilters) {
+    setSelectedFilters({
+      ...selectedFilters,
+      [type]: [],
+    });
+  }
+
+  function handleSelectFilters(
+    type: keyof SelectedFilters,
+    filters: ComboboxOption[],
+  ) {
+    setSelectedFilters({
+      ...selectedFilters,
+      [type]: filters,
+    });
+  }
+
+  const LIST_QUERY1 = useMemo(
+    () => gql`
+      query ListQuery($cursor: String) {
+        allPeople(first: 10, after: $cursor) {
+          edges {
+            node {
+              created
+              id
+              name
+              eyeColor
+              hairColor
+              skinColor
+              birthYear
+              homeworld {
+                name
+                climates
+                id
+                population
+                terrains
+              }
+              species {
+                name
+                id
+              }
+            }
+            cursor
+          }
+          pageInfo {
+            hasNextPage
+            endCursor
+          }
+          totalCount
+        }
+      }
+    `,
+    [],
+  );
+
+  const apolloClient1 = useMemo(
+    () =>
+      new ApolloClient({
+        uri: "https://swapi-graphql.netlify.app/.netlify/functions/index",
+        cache: new InMemoryCache(),
+      }),
+    [],
+  );
+
+  const { data } = useCollectionQuery<ListQueryType>({
+    query: LIST_QUERY1,
+    queryOptions: {
+      fetchPolicy: "network-only",
+      nextFetchPolicy: "cache-first",
+      client: apolloClient1,
+    },
+
+    getCollectionByPath(items) {
+      return items?.allPeople;
+    },
+  });
+
+  const items = data?.allPeople.edges || [];
+  const totalCount = data?.allPeople.totalCount || null;
+
+  const mappedData = items.map(({ node }) => ({
```

Comment:
I noticed that this doesn't actually filter the data based on the filters. Do you think it would be worth filtering the data as well?

Reply:
Correct, I didn't implement the actual filtering for 2 reasons:
1. We're not doing it in any other stories, even though technically we're exposing filters - we only show alerts or console logs.
2. I figured filtering would be done on the server side anyways, so I thought there's no point in showing how to implement it.

Given my reasoning, do you still think I should add data filtering logic?

Reply:
I think that makes sense especially when you consider that the filtering will be done server side